libs/sdk-ui-ext/src/internal/components/pluggableVisualizations/headline/headlineConfigFromInsight.ts

```diff
@@ -0,0 +1,19 @@
+// (C) 2023 GoodData Corporation
+import { IInsightDefinition, insightProperties } from "@gooddata/sdk-model";
+import { IEmbeddingCodeContext } from "../../../interfaces/VisualizationDescriptor.js";
+import { IChartConfig } from "@gooddata/sdk-ui-charts";
+
+export function headlineConfigFromInsight(
+    insight: IInsightDefinition,
+    ctx: IEmbeddingCodeContext | undefined,
+): IChartConfig {
+    if (ctx?.settings?.enableNewHeadline) {
+        const separatorsProp = ctx?.settings?.separators ? { separators: ctx?.settings?.separators } : {};
+        const comparison = insightProperties(insight)?.controls?.comparison;
```

Comment:
Comparison is optional as well but unlike separators, it will always be in the result object as undefined. I guess this is probably because we want to be able to distinguish between old headline insight created before this milestone and the ones that were created afterwards? If that is the case, please put here a comment that will explain it as it could be confusing to someone in the future why separator key is omitted when it is undefined, while comparison key is not, and he can refactor it to have it more "optimized".

Reply:
I supposed that the comparison will always exist when this milestone is rolled out with migration in place

Reply:
As the migration is not implemented yet, I will add default value for comparison to avoid potential issues